### ISSUE TITLE
`class_delegate_protocol` false positive when using `&`

### ISSUE THREAD

**Issue by Deanosaurus** [realm/SwiftLint#5982]
### New Issue Checklist

- [X] I've Updated SwiftLint to the latest version.
- [X] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

When declaring a `protocol` SwiftLint generates a false-positive for a `class_delegate_protocol` violation when `AnyObject` is added via `&`.

```swift
/// "Class Delegate Protocol Violation: Delegate protocols should be class-only so they can be weakly referenced (class_delegate_protocol)"
protocol SomeDelegate: AnyObject & Sendable {

}

/// No error.
protocol SomeDelegate: AnyObject, Sendable {

}
```

I realize technically speaking `AnyObject & Sendable` is technically a different type than independently `AnyObject` and `Sendable`, but the principle the rule is trying to get at should resolve the same.

### Command/Output

```bash
$ swiftlint --strict
```

> Class Delegate Protocol Violation: Delegate protocols should be class-only so they can be weakly referenced (class_delegate_protocol)

### Environment

* SwiftLint version: 0.57.0
* Xcode version: 16.2.0
* Installation method used: Homebrew
* Configuration file: default

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? If so, paste their
relative paths and respective contents.

- No nested config
